cmd/soroban-cli/src/config/network.rs

```diff
@@ -47,6 +47,14 @@ pub struct Args {
         help_heading = HEADING_RPC,
     )]
     pub rpc_url: Option<String>,
+    /// Optional RPC provider api key headers
+    #[arg(
+            long = "rpc-header",
+            requires = "rpc_url",
+            env = "STELLAR_RPC_HEADER",
+            help_heading = HEADING_RPC,
+        )]
+    pub rpc_header: Option<String>,
```

Comment:
I think this should probably be a multi-value key, therefore be a `Vec<String>`, but as far as I know the clap-rs `env` feature doesn't work well with it. We'd have to try it out.

Given all use cases we have only need a single header at this time, then as it is looks good, but if you wanted to do a mini timeboxed exploration on whether `Vec<String>` is supported with `env` in some way, and it turned out it worked, then I'd change it to:

```suggestion
    /// Optional RPC provider api key headers
    #[arg(
            long = "rpc-header",
            num_args(0..),
            requires = "rpc_url",
            env = "STELLAR_RPC_HEADERS",
            help_heading = HEADING_RPC,
        )]
    pub rpc_header: Vec<String>,
```

Reply:
Just re-reading your comment, and I missed the `env` part the first time. I'll timebox looking into this today 👌